docs/flow/config-language/expressions/referencing_outputs.md

```diff
@@ -1,8 +1,69 @@
 ---
 aliases:
-- /docs/agent/latest/flow/configuration-language/expressions/referencing-outputs
-title: Referencing component exports
+- /docs/agent/latest/flow/configuration-language/expressions/referencing-exports
+title: Referencing component Exports
 weight: 200
 ---
 
-# Types and Values
+# Referencing component Exports
+Referencing component Exports is what enables River to dynamically configure
+and connect components using expressions.
+
+While components can work in isolation, they can be much more versatile when
+one component's behavior and data flow is bound to the Exports of another,
+building a dependency relationship between the two.
+
+## Using references
+These references are built by combining the component's name, label and named
+Export with dots.
+
+For example, the contents of a file exported by the `local.file` component
+might be referenced as `local.file.targets.content`, while a
+`prometheus.remote_write` component instance might expose a receiver for
+metrics like `prometheus.remote_write.onprem.receiver`.
+
+Let's see that in action:
+```river
+local.file "targets" {
+	filename = "/etc/agent/targets" 
+}
+
+prometheus.scrape "default" {
+	targets    = [{ "__address__" = local.file.target.content }] 
+	forward_to = [prometheus.remote_write.default.receiver]
+}
+
+prometheus.remote_write "onprem" {
+	remote_write {
+		url = "http://prometheus:9009/api/prom/push"
+	}
+}
+```
+
+In the previous example, we managed to wire together a very simple pipeline by
+writing a few River expressions.
+```
```

Comment:
Do we want to keep this kind of diagrams across the docs? Do they render properly in our docs? else we could use mermaid or something like that

Reply:
Not sure if the diagrams are a good idea, looking for feedback on that :) 

I just want to make sure that the concept of referencing outputs is well-explained.

Reply:
I just checked with a mermaid fenced codeblock, but it only seems to be rendered in Github, it doesn't show when browsing after running `make docs`. So let's either keep the ASCII graph or get rid of it altogether.